src/helpers/squidTransactions.ts

```diff
@@ -0,0 +1,145 @@
+import axios from 'axios';
+import { ethers } from 'ethers';
+import config from '@/config/configuration';
+const integratorId: string = config.SQUID_INTEGRATOR_ID;
+
+export type SquidTokenType = {
+  symbol: string;
+  address: string;
+  chainId: string;
+  name: string;
+  decimals: number;
+  usdPrice: number;
+  coingeckoId: string;
+  type: string;
+  logoURI: string;
+  subGraphIds: string[];
+  subGraphOnly: boolean;
+  active: boolean;
+  balance?: number | undefined;
+};
+
+export const getRoute = async (params: any) => {
+  try {
+    const result = await axios.post(
+      'https://apiplus.squidrouter.com/v2/route',
+      params,
+      {
+        headers: {
+          'x-integrator-id': integratorId,
+          'Content-Type': 'application/json',
+        },
+      },
+    );
+    const requestId = result.headers['x-request-id'];
+    return { data: result.data, requestId: requestId };
+  } catch (error: any) {
+    if (error.response) {
+      console.error('API error:', error.response.data);
+    }
+    console.error('Error with parameters:', params);
+    throw error;
+  }
+};
+
+export const approveSpending = async (
+  transactionRequestTarget: string,
+  fromToken: string,
+  fromAmount: string,
+) => {
+  const erc20Abi = [
+    'function approve(address spender, uint256 amount) public returns (bool)',
+  ];
+
+  let provider = new ethers.BrowserProvider(window.ethereum);
+  let signer = await provider.getSigner();
+
+  const tokenContract = new ethers.Contract(fromToken, erc20Abi, signer);
+
+  try {
+    const tx = await tokenContract.approve(
+      transactionRequestTarget,
+      fromAmount,
+    );
+    await tx.wait();
+    console.log(
+      `Approved ${fromAmount} tokens for ${transactionRequestTarget}`,
+    );
+  } catch (error) {
+    console.error('Approval failed:', error);
+    throw error;
+  }
+};
+
+export const getStatus = async (params: any) => {
+  try {
+    const result = await axios.get(
+      'https://apiplus.squidrouter.com/v2/status',
+      {
+        params: {
+          transactionId: params.transactionId,
+          requestId: params.requestId,
+          fromChainId: params.fromChainId,
+          toChainId: params.toChainId,
+        },
```

Comment:
you can use params directly:
```suggestion
        params,
```